app/models/spree/ability.rb

```diff
@@ -274,8 +283,37 @@ def add_order_cycle_management_abilities(user)
     end
 
     def add_order_management_abilities(user)
-      can [:index, :create], Spree::Order
-      can [:read, :update, :fire, :resend, :invoice, :print], Spree::Order do |order|
+      can [:manage_order_sections], Spree::Order do |order|
+        user.admin? ||
+          order.distributor.nil? ||
+          user.enterprises.include?(order.distributor) ||
+          order.order_cycle&.coordinated_by?(user)
+      end
+
+      can [:edit_as_producer_only], Spree::Order do |order|
+        cannot?(:manage_order_sections, order) && can_edit_as_producer(order, user)
+      end
+
+      can [:index], Spree::Order do |order|
+        user.admin? ||
+          user.enterprises.any?(&:is_distributor) ||
+          can_edit_as_producer(order, user)
+      end
```

Comment:
I'm not understanding this rule. The `index` action is to show a list of orders, so I don't think any order object would be passed in. Am I understanding it right?

If so, instead of `can_edit_as_producer(order, user)`, I would expect something like "can edit any of my supplied orders". 
We probably don't need to look at orders. How about "do any enterprises distribute my products, and do they allow editing orders?"

```suggestion
      can [:index], Spree::Order
        user.admin? ||
          user.enterprises.any?(&:is_distributor) ||
          user.enterprises.distributors.where(enable_producers_to_edit_orders: true).exist?
      end
```
(I'm not sure if that is correct, but hopefully the tests will show!)